src/core/toga/command.py

```diff
@@ -106,7 +108,7 @@ def icon(self, icon_or_name):
         if isinstance(icon_or_name, Icon) or icon_or_name is None:
             self._icon = icon_or_name
         else:
-            self._icon = Icon(icon_or_name)
+            self._icon = Icon(os.getcwd() + "/" + icon_or_name)
```

Comment:
This may fix a bug you're seeing, but it's the wrong approach in this case. 

The name passed to `Icon` is an "icon identifier", not a filename. A filename *can* act as an icon identifier, but that shouldn't be the only interpretation. The icon identifier encompasses finding the right file extension for a given platform (.e.g, `.ico` vs `.icns`), as well as any size prefix that is required (e.g., GTK allows for icons at multiple sizes, so the icon `icons/brutus` might be rendered using `brutus-32.png` or `brutus-72.png`, depending on circumstances).

The name itself is interpreted relative to the *app* folder - i.e., the folder that declares the `__main__` entry point. so - the `icons/brutus` in tutorial 2 should be sufficient to find the icons folder in that app. The only exception to this is if you pass in the `system` flag when constructing the Icon; if `system=True`, the path is relative to the toga-core module. This means `Icon('resources/toga', system=True)` should find the Toga "yak" icon.

So - in this case, if icons aren't loading, the problem is with GTK icon path resolution, not the path being used constructing the icon. As a guide - the macOS implementation correctly finds the icons *without* the change you've proposed here.

Reply:
In regard to icons, as I was browsing through the code, I found that the code depends on finding the location of the images on several things, which including that the `helloworld` in command `python -m helloworld` must be a folder not a `Python` file, `helloworld` treated as a module by using `-m` flag when executing the program, and `importlib_metadata.metadata` is defined, otherwise `Tog` will be automatically called my `app_name` and also the `module_name` as `toga` rather than `helloworld` and this will miss the execution the path to icons folder.

But because `Toga` is primarily treated as part of `BeeWare` project, which implicitly define the previous parameters correctly while creating the app, I tried the second lesson of `Toga` with `BeeWare` and succeeded in finding the icons. 

So, what do you think?